rs/state_machine_tests/src/lib.rs

```diff
@@ -751,19 +752,20 @@ pub struct StateMachineNode {
     pub committee_signing_key: ic_crypto_ed25519::PrivateKey,
     pub dkg_dealing_encryption_key: ic_crypto_ed25519::PrivateKey,
     pub idkg_mega_encryption_key: ic_crypto_ed25519::PrivateKey,
+    pub http_ip_addr: Ipv6Addr,
+    pub xnet_ip_addr: Ipv6Addr,
 }
 
-impl From<u64> for StateMachineNode {
-    fn from(i: u64) -> Self {
-        let mut bytes = [0; 32];
-        bytes[..8].copy_from_slice(&(4 * i).to_le_bytes());
-        let node_signing_key = ic_crypto_ed25519::PrivateKey::deserialize_raw_32(&bytes);
-        bytes[..8].copy_from_slice(&(4 * i + 1).to_le_bytes());
-        let committee_signing_key = ic_crypto_ed25519::PrivateKey::deserialize_raw_32(&bytes);
-        bytes[..8].copy_from_slice(&(4 * i + 2).to_le_bytes());
-        let dkg_dealing_encryption_key = ic_crypto_ed25519::PrivateKey::deserialize_raw_32(&bytes);
-        bytes[..8].copy_from_slice(&(4 * i + 3).to_le_bytes());
-        let idkg_mega_encryption_key = ic_crypto_ed25519::PrivateKey::deserialize_raw_32(&bytes);
+impl StateMachineNode {
+    fn new(rng: &mut StdRng) -> Self {
+        let node_signing_key = ic_crypto_ed25519::PrivateKey::deserialize_raw_32(&rng.gen());
+        let committee_signing_key = ic_crypto_ed25519::PrivateKey::deserialize_raw_32(&rng.gen());
+        let dkg_dealing_encryption_key =
+            ic_crypto_ed25519::PrivateKey::deserialize_raw_32(&rng.gen());
+        let idkg_mega_encryption_key =
+            ic_crypto_ed25519::PrivateKey::deserialize_raw_32(&rng.gen());
+        let http_ip_addr = Ipv6Addr::from(rng.gen::<[u16; 8]>());
+        let xnet_ip_addr = Ipv6Addr::from(rng.gen::<[u16; 8]>());
```

Comment:
There's no guarantee that randomly generated numbers will be distinct. Why not use something like a counter?

Or am I missing something?

Reply:
It isn't guaranteed, but I believe it is very unlikely to get the same `[u16; 8]`, i.e., 128 bits.

Reply:
Oh sorry, I read that as `[<random_u16>; 8]`. My bad.